examples/ablation-workshop-mpi.py

```diff
@@ -0,0 +1,763 @@
+        comm.Barrier()
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    pressure_boundaries = {
+        BoundaryDomainTag("prescribed"):
+            DirichletDiffusionBoundary(101325),
+        BoundaryDomainTag("neumann"):
+            NeumannDiffusionBoundary(0.0)
+    }
+
+    def my_presc_bdry(u_minus):
+        return +u_minus
+
+    def my_wall_bdry(u_minus):
+        return -u_minus
+
+    velocity_boundaries = {
+        BoundaryDomainTag("prescribed"):
+            my_presc_bdry,
+        BoundaryDomainTag("neumann"):
+            my_wall_bdry
+    }
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    # {{{ Initialize wall model
+
+    from mirgecom.materials.tacot import WallTabulatedEOS
+    import mirgecom.materials.tacot as my_composite
+
+    my_material = my_composite.SolidProperties()
+    my_gas = my_composite.GasProperties()
+    pyrolysis = my_composite.Pyrolysis()
+
+#    # FIXME there must be a way to get the table from the "materials" directory
+    bprime_table = \
+        (np.genfromtxt("aw_Bprime.dat", skip_header=1)[:, 2:6]).reshape((25, 151, 4))
+    bprime_class = my_composite.BprimeTable(table=bprime_table)
+
+    wall_model = WallTabulatedEOS(wall_material=my_material)
+
+    # }}}
+
+    from mirgecom.gas_model import GasModel
+    gas_model = GasModel(eos=my_gas, wall=wall_model)
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    # soln setup and init
+    wall_density = np.empty((3,), dtype=object)
+
+    wall_density[0] = 30.0 + nodes[0]*0.0
+    wall_density[1] = 90.0 + nodes[0]*0.0
+    wall_density[2] = 160. + nodes[0]*0.0
+    temperature = 300.0 + nodes[0]*0.0
+    pressure = 101325.0 + nodes[0]*0.0
+
+    pressure = force_evaluation(actx, pressure)
+    temperature = force_evaluation(actx, temperature)
+    wall_density = force_evaluation(actx, wall_density)
+
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    if restart_file:
+        t = restart_data["t"]
+        istep = restart_data["step"]
+        cv = restart_data["cv"]
+        wall_density = restart_data["wall_density"]
+    else:
+        t = 0
+        istep = 0
+        import mirgecom.multiphysics.phenolics as phenolics
+        cv = phenolics.initializer(
+            dcoll=dcoll, gas_model=gas_model,
+            wall_density=wall_density,
+            pressure=pressure, temperature=temperature)
+
+    temperature_seed = nodes[0]*0.0 + 300.0
+
+    # stand-alone version of the "gas_model"to bypass some unnecessary variables
+    # for this particular case
+    def _make_state(cv, gas_model, temperature_seed, wall_density):
+        """Return the fluid+wall state for porous media flow.
+
+        Ideally one would use the gas_model.make_fluid_state but, since this
+        case use tabulated data and equilibrium gas assumption, it was
+        implemented in this stand-alone function. Note that some functions
+        have slightly different calls and the absence of species.
+        """
```

Comment:
As we talked, I don't think it is worth to change anything in mirge-com because of this specific case. It is a sanity check setup that I can use to compare against literature results that employ "exactly" the steps I implemented here.
Practical predictions cases will use either a more simplified approach (on-going progress) or the full physics (future work), but such implementations will be more natural with what we have right now.